src/control/xml/DoubleArrayAttribute.cpp

```diff
@@ -1,31 +1,29 @@
 #include "DoubleArrayAttribute.h"
 #include "Util.h"
 
-DoubleArrayAttribute::DoubleArrayAttribute(const char* name, double* values, int count) : XMLAttribute(name)
+DoubleArrayAttribute::DoubleArrayAttribute(const char* name, std::vector<double>&& values)
+ : XMLAttribute(name)
+ , values(std::move(values))
 {
-	this->values = values;
-	this->count = count;
 }
 
 DoubleArrayAttribute::~DoubleArrayAttribute()
 {
-	delete values;
-	values = nullptr;
 }
 
 void DoubleArrayAttribute::writeOut(OutputStream* out)
 {
-	if (this->count > 0)
+	if (!this->values.empty())
 	{
 		char str[G_ASCII_DTOSTR_BUF_SIZE];
 		// g_ascii_ version uses C locale always.
 		g_ascii_formatd(str, G_ASCII_DTOSTR_BUF_SIZE, Util::PRECISION_FORMAT_STRING, this->values[0]);
 		out->write(str);
 
-		for (int i = 1; i < this->count; i++)
+		for (auto it = this->values.begin() + 1; it != this->values.end(); it++)
 		{
 			// g_ascii_ version uses C locale always.
-			g_ascii_formatd(str, G_ASCII_DTOSTR_BUF_SIZE, Util::PRECISION_FORMAT_STRING, this->values[i]);
+			g_ascii_formatd(str, G_ASCII_DTOSTR_BUF_SIZE, Util::PRECISION_FORMAT_STRING, *it);
```

Comment:
I don't know what happens if we use `std::to_string()` or `std::to_chars` here, but it could be cleaner. And both should use the c locale.

Reply:
The alternative is to use `std::sprintf` but it's no less concise than this.